config/config_test.go

```diff
@@ -66,7 +67,92 @@ func (s *ConfigTestSuite) TestNew() {
 		c, err := New(cfg)
 		s.NoError(err)
 
-		s.Equal(test.expectedPatterns, c.Streams[0].DataYaml.ActivatedRules.CISK8S)
+		s.Equal(test.expected, c.Streams[0].DataYaml.ActivatedRules.CISK8S)
+	}
+}
+
+func (s *ConfigTestSuite) TestUpdate() {
+	config := `
+    streams:
+      - data_yaml:
+          activated_rules:
+            cis_k8s:
+              - a
+              - b
+              - c
+              - d
+              - e`
+
+	var tests = []struct {
+		update   string
+		expected []string
+	}{
+		{
+			`
+          streams:
+            - data_yaml:
+                activated_rules:
+                  cis_k8s:
+                    - a
+                    - b
+                    - c
+                    - d
+        `,
+			[]string{"a", "b", "c", "d"},
+		},
+		{
+			`
+          streams:
+            - data_yaml:
+                activated_rules:
+                  cis_k8s:
+                    - b
+                    - c
+                    - d
+                    - e
+                    - f
+        `,
+			[]string{"b", "c", "d", "e", "f"},
+		},
+		{
+			`
+          streams:
+            - data_yaml:
+                activated_rules:
+                  cis_k8s: []
+        `,
+			[]string{},
+		},
+		{
+			`
+          streams:
+            - data_yaml:
+                activated_rules:
+                  cis_k8s:
+                    - a
+                    - b
+                    - c
+                    - d
+                    - e
+        `,
+			[]string{"a", "b", "c", "d", "e"},
+		},
+	}
+
+	cfg, err := common.NewConfigFrom(config)
+	s.NoError(err)
+
+	c, err := New(cfg)
+	s.NoError(err)
+
+	for _, test := range tests {
+		cfg, err := common.NewConfigFrom(test.update)
+		s.NoError(err)
+
+		err = c.Update(cfg)
+		s.NoError(err)
+
+		s.Equal(test.expected, c.Streams[0].DataYaml.ActivatedRules.CISK8S)
```

Comment:
Let's make sure that other config fields don't get overridden by mistake

Reply:
Have added tests for that.